google/gax/__init__.py

```diff
@@ -482,3 +489,160 @@ def __next__(self):
         if self._index >= len(self._current):
             self._current = None
         return resource
+
+
+def _from_any(pb_type, any_pb):
+    if not any_pb.Is(pb_type.DESCRIPTOR):
+        raise TypeError(
+            'Could not convert {} to {}'.format(
+                any_pb.__class__.__name__, pb_type.__name__))
+
+    return pb_type.FromString(any_pb.value)
+
+
+class ResultError(GaxError):
+    """Thrown when an operation completes with an error result.
+    """
+
+    @classmethod
+    def from_operation_error(cls, operation_error):
+        """Factory: construct instance from an operation error result.
+
+        Args:
+            operation_error (google.rpc.Status): the error result of the
+                operation.
+        """
+        return cls(operation_error.message, operation_error.code)
+
+    def __init__(self, msg, code=None):
+        """Constructor.
+
+        Args:
+            msg (string): describes the error that occurred.
+            code (int, optional): the status code, which should be an enum
+                value of google.rpc.Code.
+        """
+        super(ResultError, self).__init__(msg)
+        self.code = code
+
+    def __str__(self):
+        msg = super(ResultError, self).__str__()
+        return 'ResultError({}, with code {})'.format(msg, self.code)
+
+
+class OperationFuture(object):
+    """A Future which polls a service for completion via OperationsApi."""
+
+    def __init__(self, operation, client, result_type, metadata_type,
+                 call_options=None):
+        """Constructor.
+
+        Args:
+            operation (google.longrunning.Operation): the initial long-running
+                operation.
+            client (google.cloud.gapic.longrunning.operations_api.OperationsApi):
```

Comment:
@eoogbe is correct. This GAX code is a library that we publish ([`google-gax`](https://pypi.python.org/pypi/google-gax)), and it's already a dependency for the generated GAPIC code for each API.

For auto-gen, I generally copy the code from `artman/output/gcloud-...` to the repo where I want it using `cp -rf artman/output/gcloud-... target_dir`. That shouldn't clobber anything except for conflicting filenames (of which there shouldn't be any).

BTW, you only need the `google/` sub-directory from the generated GAPIC LRO code. The rest is package meta-data that we don't need, since `gax-python` already has its own package metadata. However, you should check that we have all the same dependencies, path declarations, etc. from `setup.py`.